### PR TITLE
To Add Note Section in Matcher

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: Refer #18633. discussion with @rojkov, adding Note Section in Matcher doc for better clarity of "`max_program_size`" uses
part of #18633

### CODE REVIEW

**Comment by ankatare** [comment]
@rojkov @snowp FYI Please.

File: api/envoy/type/matcher/v3/regex.proto [hunk 1]
```diff
@@ -45,6 +45,8 @@ message RegexMatcher {
     //
     // This field is deprecated; regexp validation should be performed on the management server
     // instead of being done by each individual client.
+.. note::
```
Comment (phlax) [inline api/envoy/type/matcher/v3/regex.proto:48]: @ankatare im assuming this should have been indented inside the comment, no ?

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto:48]: @phlax Ahh. Sorry. 
it should be like...  as below ?  
//.. note::
  //   The program size is still checked against the runtime value.

Comment (phlax) [reply api/envoy/type/matcher/v3/regex.proto:48]: i think the indent *of* the comment should be the same, the `.. note::` probs requires a single space indent, and the following line, not sure, but i think 4 indents

you might need another comment line underneath, also not sure about that

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto:48]: humm, let me check and run first it locally.. Thanks

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto]: Done

**Comment by ankatare** [comment]
> 

nice catch, Thnaks

**Comment by ankatare** [comment]
> docs rendered here https://storage.googleapis.com/envoy-pr/19260/docs/index.html
> 
> current rendered change is here https://storage.googleapis.com/envoy-pr/0e59bea/docs/api-v3/type/matcher/v3/regex.proto.html#type-matcher-v3-regexmatcher-googlere2
> 
> @ankatare the formatting is still not correct - probs requires at least a space before i think

@phlax eager to know, how you check docs generated doc in runtime ?

**Comment by phlax** [comment]
depending on how much/type of change either by building or submitting and checking ci

**Comment by ankatare** [comment]
> depending on how much/type of change either by building or submitting and checking ci

i mean how you checked https://storage.googleapis.com/envoy-pr/0e59bea/docs/api-v3/type/matcher/v3/regex.proto.html#type-matcher-v3-regexmatcher-googlere2  path runtime ?

**Comment by phlax** [comment]
oh i see - i created the first link manually, let it resolve, and then searched the (api) docs for the changed file

**Comment by ankatare** [comment]
> oh i see - i created the first link manually, let it resolve, and then searched the (api) docs for the changed file

Nice.. i got it now...  :) 
https://storage.googleapis.com/envoy-pr/155733c/docs/api-v3/type/matcher/v3/regex.proto.html

**Review by phlax — DISMISSED** [review]
lgtm, from docs pov, needs api approval

cc @adisuissa

File: api/envoy/type/matcher/v3/regex.proto [hunk 2]
```diff
@@ -45,6 +45,11 @@ message RegexMatcher {
     //
     // This field is deprecated; regexp validation should be performed on the management server
     // instead of being done by each individual client.
+    //
+    // .. note::
+    //
+    //  The program size is still checked against the runtime value.
```
Comment (adisuissa) [inline api/envoy/type/matcher/v3/regex.proto:51]: I think this note can be clarified a bit. What do you think of the following:
```suggestion
    //  Although this field is deprecated, the program size will still be checked against the *max_program_size* runtime value.
```

**Review by adisuissa — COMMENTED** [review]
Thanks for clarifying this!

File: api/envoy/type/matcher/v3/regex.proto (hunk 2 above)
Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto]: Sure

File: api/envoy/type/matcher/v3/regex.proto [hunk 3]
```diff
@@ -48,7 +48,7 @@ message RegexMatcher {
     //
     // .. note::
     //
-    //  The program size is still checked against the runtime value.
+    //  Although this field is deprecated, the program size will still be checked against the *max_program_size* runtime value.
```
Comment (phlax) [inline api/envoy/type/matcher/v3/regex.proto:51]: max_program_size is an inline literal and needs a double backtick - ie ` ``max_program_size `` `

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto]: Oh Ok

**Comment by ankatare** [comment]
@adisuissa @phlax PTAL

**Review by adisuissa — DISMISSED** [review]
/lgtm api
Thanks!

File: api/envoy/type/matcher/v3/regex.proto [hunk 4]
```diff
@@ -45,6 +45,12 @@ message RegexMatcher {
     //
     // This field is deprecated; regexp validation should be performed on the management server
     // instead of being done by each individual client.
+    //
+    // .. note::
+    //
+    //  Although this field is deprecated, the program size will still be checked against the
+    //  ``max_program_size`` runtime value.
```
Comment (rojkov) [inline api/envoy/type/matcher/v3/regex.proto:52]: It's better to use a fully qualified name.
```suggestion
    //  gloabl ``re2.max_program_size.error_level`` runtime value.
```
Bear in mind there's also `re2.max_program_size.warn_level` (effectively not set by default). If the compiled regex is greater than this then a warning is issued, but the process keeps running.

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto:52]: Ok

File: api/envoy/type/matcher/v3/regex.proto [hunk 5]
```diff
@@ -45,6 +45,12 @@ message RegexMatcher {
     //
     // This field is deprecated; regexp validation should be performed on the management server
     // instead of being done by each individual client.
+    //
+    // .. note::
+    //
+    //  Although this field is deprecated, the program size will still be checked against the
+    //  gloabl ``re2.max_program_size.error_level`` runtime value.
```
Comment (rojkov) [inline api/envoy/type/matcher/v3/regex.proto:52]: Sorry, I made a typo here.

Comment (ankatare) [reply api/envoy/type/matcher/v3/regex.proto]: Done,

**Review by rojkov — APPROVED** [review]
Thank you for the clarification! LGTM

**Comment by phlax** [comment]
@adisuissa can we get final api-shep signoff, ta

**Review by adisuissa — APPROVED** [review]
/lgtm api

**Pull request merged**